diff --git a/lib/server/parse.js b/lib/server/parse.js
index bf41fa29816f1975bde9a64f6f685f8bd6d8ae3a..fdcf40ee01e71c0fbf742822cc585dc3bf6677dc 100644
--- a/lib/server/parse.js
+++ b/lib/server/parse.js
@@ -109,7 +109,10 @@ function html2text(html, type, url = "?") {
                 // <a class="hash-link" href="#first-header" title="Direct link to heading">#</a>
                 .not("a[aria-hidden=true], a.hash-link")
                 .text();
-            const hash = $(heading).find("a.hash-link").attr("href") || "";
+            // const hash = $(heading).find("a.hash-link").attr("href") || "";
+            const linkHash = $(heading).find("a.hash-link").attr("href") || "";
+            const [, hashPart] = linkHash.split("#");
+            const hash = hashPart ? `#${hashPart}` : "";
             let $sectionElements;
             if ($(heading).parents(".markdown").length === 0) {
                 // $(heading) is the page title
